Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -1168,6 +1168,31 @@ void PcapLiveDevice::setDefaultGateway()
 #endif
 }
 
+std::vector<IPAddress> PcapLiveDevice::getIPAddresses() const
+{
+	std::vector<IPAddress> results;
+	for (const auto& address : m_Addresses)
```

Comment:
it might be simpler to use `sockaddr2string` to convert the address to string, and the call `IPAddress` c'tor with this string

Reply:
Eh, wouldn't that not achieve the same but with extra steps? We add encoding to string and then decoding from string for not much benefit. Atm, most operations are read/check operations.

Maybe it would be more concise and skip the if checks in the current function, but I don't think the current implementatios is that bad.

Reply:
What I mean is that you can do something like this which is shorter and more concise:
```cpp
for (const auto& address : m_Addresses)
{
	std::array<char, INET6_ADDRSTRLEN> addrAsString;
	internal::sockaddr2string(addrIter.addr, addrAsString.data(), addrAsString.size());
	try
	{
		results.push_back(IPAddress(addrAsString));
	}
	catch(...) {}
}
```

Reply:
True, but looking at the way it will be implemented adds a lot of overhead that is avoided by the current code.

This is  the current implementation of the `IPAddress` string constructor.
```
IPAddress::IPAddress(const std::string& addrAsString)
{
	if (IPv4Address::isValidIPv4Address(addrAsString))  //  <-- string parse
	{
		m_Type = IPv4AddressType;
		m_IPv4 = IPv4Address(addrAsString);  //  <-- potential string parse
	}
	else if (IPv6Address::isValidIPv6Address(addrAsString))  //  <-- string parse
	{
		m_Type = IPv6AddressType;
		m_IPv6 = IPv6Address(addrAsString);  //  <-- potential string parse
	}
	else
	{
		throw std::invalid_argument("Not a valid IP address: " + addrAsString);
	}
}
```
This is between 2 and 3 parses of the string address to get the IP object in addition to the initial encode of the IP to a string. The current implementation, while slightly longer is at most 4 nullptr checks, 2 integer equality checks and a buffer copy.

Reply:
ok, we can go with your approach 🙂